pkg/client/unversioned/testclient/testclient.go

```diff
@@ -348,3 +338,34 @@ func (c *FakeExperimental) Jobs(namespace string) client.JobInterface {
 func (c *FakeExperimental) Ingress(namespace string) client.IngressInterface {
 	return &FakeIngress{Fake: c, Namespace: namespace}
 }
+
+type FakeDiscovery struct {
+	*Fake
+}
+
+func (c *FakeDiscovery) ServerResourcesForGroupVersion(groupVersion string) (*unversioned.APIResourceList, error) {
+	action := ActionImpl{
+		Verb:     "get",
+		Resource: "resource",
+	}
+	c.Invokes(action, nil)
+	resource, found := c.Resources[groupVersion]
+	if found {
```

Comment:
You could just `return c.Resources[groupVersion], nil` because the default value for a missing value will be nil.